src/Install/Requirements.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains Class Requirements.
+ *
+ * This file is about checks for requirements during install of MongoDB module.
+ */
+
+namespace Drupal\mongodb\Install;
+
+/**
+ * Class Requirements implements hook_requirements().
+ */
+class Requirements {
+  public function __construct() {
+    $this->extension_name = '';
```

Comment:
Better move that to properties on the class.

Also, since mongodb_requirements() uses check() as a static method, the constructor won't be run. mongodb_requirements() should probably use an instance of the class and check() as an instance method so that the properties are initialized.